tools/get_coverage_files.bzl

```diff
@@ -0,0 +1,82 @@
+load("//cc:defs.bzl", "LIBRARY", "TEST_LIBRARY")
+load("@bazel_skylib//rules:common_settings.bzl", "BuildSettingInfo")
+
+FilesInfo = provider(
+    fields = {
+        "files": "files of the target",
+    },
+)
+
+def _get_cov_files_impl(target, ctx):
+    files = []
+
+    if not CcInfo in target:
+        return [FilesInfo(files = [])]
+
+    tags = getattr(ctx.rule.attr, "tags", [])
+    if not LIBRARY in tags and not TEST_LIBRARY in tags:
```

Comment:
Shouldn't we also grab files from targets that are tag `BINARY`?

Reply:
Yes, we should. Sorry, I removed it when I was working on the new design.